src/main/java/org/spongepowered/api/item/recipe/ShapedRecipe.java

```diff
@@ -52,85 +61,40 @@
     int getHeight();
 
     /**
-     * Gets the ingredient required in the given coordinates.
+     * Gets the ingredient required by the given symbol.
      *
-     * @param x The x coordinate
-     * @param y The y coordinate
+     * @param symbol The ingredient symbol
      * @return The ingredient
      */
-    Optional<ItemStack> getIngredient(int x, int y);
-
-    /**
-     * Gets the ingredient required in the given position.
-     *
-     * @param pos The position
-     * @return The ingredient
-     */
-    Optional<ItemStack> getIngredient(Vector2i pos);
+    Optional<ItemStack> getIngredient(char symbol);
 
     interface Builder extends ResettableBuilder<ShapedRecipe, Builder> {
-        /**
-         * Sets the width of the grid for the ShapedRecipe.
-         *
-         * @param width The width of the grid
-         * @return fluent interface
-         */
-        Builder width(int width);
-
-        /**
-         * Sets the height of the grid for the ShapedRecipe.
-         *
-         * @param height The height of the grid
-         * @return fluent interface
-         */
-        Builder height(int height);
 
         /**
-         * Sets the dimensions of the grid for the ShapedRecipe in one method call.
+         * Sets the aisle pattern for the ShapedRecipe.
          *
-         * @param dimensions The dimensions of the grid
+         * @param aisle A string array of ingredients
          * @return fluent interface
          */
-        Builder dimensions(Vector2i dimensions);
+        Builder aisle(String... aisle);
 
         /**
-         * Sets the ingredient required by the recipe in the given coordinates.
+         * Sets an ingredient based on the pattern for the ShapedRecipe.
          *
-         * @param x The x coordinate
-         * @param y The y coordinate
+         * @param symbol The ingredient symbol
          * @param ingredient The ingredient to set, or remove if null
          * @return fluent interface
          */
-        Builder ingredient(int x, int y, @Nullable ItemStack ingredient);
-
-        /**
-         * Sets the ingredient required by the recipe in the given position.
-         *
-         * @param pos The position
-         * @param ingredient The ingredient to set, or remove if null
-         * @return fluent interface
-         */
-        Builder ingredient(Vector2i pos, @Nullable ItemStack ingredient);
-
-        /**
-         * Sets the ingredients required by the recipe in the given row.
-         *
-         * @param row The number of the row
-         * @param ingredients A list of ItemStacks to set as ingredients. If one
-         *                    of them is null the ingredient in that position is
-         *                    not added.
-         * @return fluent interface
-         */
-        Builder row(int row, ItemStack... ingredients);
+        Builder where(char symbol, @Nullable ItemStack ingredient);
```

Comment:
Declare a `throws` contract for if the specified symbol is not defined in the aisle, even if the exception type is unchecked.